pkg/standalone-utils/test/foundry/BalancerFeeBurner.t.sol

```diff
@@ -0,0 +1,259 @@
+
+    address daiWethPool;
+    address wethUsdcPool;
+
+    function setUp() public override {
+        BaseVaultTest.setUp();
+
+        orderDeadline = block.timestamp + ORDER_LIFETIME;
+        feeBurner = new BalancerFeeBurner(vault);
+        (feeRecipient, ) = makeAddrAndKey("feeRecipient");
+
+        feeSweeper = new ProtocolFeeSweeper(vault, feeRecipient);
+
+        feeBurnerAuth = IAuthentication(address(feeBurner));
+        feeSweeperAuth = IAuthentication(address(feeSweeper));
+
+        authorizer.grantRole(feeSweeperAuth.getActionId(IProtocolFeeSweeper.setFeeRecipient.selector), admin);
+        authorizer.grantRole(feeSweeperAuth.getActionId(IProtocolFeeSweeper.setTargetToken.selector), admin);
+        authorizer.grantRole(feeSweeperAuth.getActionId(IProtocolFeeSweeper.addProtocolFeeBurner.selector), admin);
+        authorizer.grantRole(feeSweeperAuth.getActionId(IProtocolFeeSweeper.sweepProtocolFeesForToken.selector), admin);
+
+        // Allow the fee sweeper to withdraw protocol fees.
+        authorizer.grantRole(
+            IAuthentication(address(feeController)).getActionId(
+                IProtocolFeeController.withdrawProtocolFeesForToken.selector
+            ),
+            address(feeSweeper)
+        );
+
+        // Allow the fee sweeper to burn.
+        authorizer.grantRole(feeBurnerAuth.getActionId(IProtocolFeeBurner.burn.selector), address(feeSweeper));
+        authorizer.grantRole(feeBurnerAuth.getActionId(IProtocolFeeBurner.burn.selector), alice);
+
+        authorizer.grantRole(feeBurnerAuth.getActionId(IBalancerFeeBurner.setBurnPath.selector), alice);
+
+        vm.prank(admin);
+        feeSweeper.addProtocolFeeBurner(feeBurner);
+
+        // Craate dai -> weth -> usdc path
+        (daiWethPool, ) = _createPool([address(dai), address(weth)].toMemoryArray(), "pool");
+        (wethUsdcPool, ) = _createPool([address(weth), address(usdc)].toMemoryArray(), "pool");
+
+        approveForPool(IERC20(daiWethPool));
+        approveForPool(IERC20(wethUsdcPool));
+
+        vm.startPrank(lp);
+        _initPool(daiWethPool, [poolInitAmount, poolInitAmount].toMemoryArray(), 0);
+        _initPool(wethUsdcPool, [poolInitAmount, poolInitAmount].toMemoryArray(), 0);
+        vm.stopPrank();
+    }
+
+    function testSweepAndBurn() public {
+        IBalancerFeeBurner.SwapPathStep[] memory steps = new IBalancerFeeBurner.SwapPathStep[](1);
+        steps[0] = IBalancerFeeBurner.SwapPathStep({ pool: pool, tokenOut: usdc });
+
+        vm.prank(alice);
+        feeBurner.setBurnPath(dai, steps);
+
+        // Set up the sweeper to be able to burn.
+        vm.startPrank(admin);
+        feeSweeper.setTargetToken(usdc);
+
+        // Put some fees in the Vault.
+        vault.manualSetAggregateSwapFeeAmount(pool, dai, DEFAULT_AMOUNT);
+
+        vm.expectEmit();
+        emit IProtocolFeeBurner.ProtocolFeeBurned(pool, dai, DEFAULT_AMOUNT, usdc, DEFAULT_AMOUNT, feeRecipient);
```

Comment:
Can we also check balances before and after to make sure the recipient received the output, and the vault consumed the input?